### PR TITLE
Build Main Dashboard Case Card Component

### PR DESCRIPTION
**Pull request by JennyVong**
## Notion ticket link

[Main Dashboard Case Cards](https://www.notion.so/uwblueprintexecs/134e1f159caa4f73939d11a700c325ad?v=39cf7d5a98384eb9af2bff2265d0ac8a&p=819164235311446e843fa6b76fcb3b9b&pm=s)

## Implementation description
* make a new component (in a new /component/dashboard folder) for this card so it can be reusable.
* export a new type for all the Case details, and this type should be passed as a prop into our new component.
* The View all icon does not perform an action for now.

## Steps to test
1. pull branch and `import CaseCard from "../dashboard/CaseCard";` into `HomePage.tsx`
2. hardcode dummy card into homepage:
   Pending case: 
    `<CaseCard
        caseTitle="Case 1"
        caseLead="Case Lead"
        date="28/05/2023"
        familyName="family name"
        caseTag="PENDING"
      />`
   Active case: 
    `<CaseCard
        caseTitle="Case 1"
        caseLead="Case Lead"
        date="28/05/2023"
        familyName="family name"
        caseTag="ACTIVE"
      />`
   Submitted case: 
    `<CaseCard
        caseTitle="Case 1"
        caseLead="Case Lead"
        date="28/05/2023"
        familyName="family name"
        caseTag="SUBMITTED"
      />`
   Archived case: 
    `<CaseCard
        caseTitle="Case 1"
        caseLead="Case Lead"
        date="28/05/2023"
        familyName="family name"
        caseTag="ARCHIVED"
      />`

Screenshot 2023-05-28 at 12 07 19 PM
Screenshot 2023-05-28 at 12 07 49 PM
Screenshot 2023-05-28 at 12 08 10 PM
Screenshot 2023-05-28 at 12 08 22 PM

## What should reviewers focus on?
* The card itself does not have a fixed size (i think this is done when formatting cards onto homepage?) 
* case tag should have different colours based on their status
* all fields should be displayed when passing in CaseCard props
* hover effect on cards

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/dashboard/CaseCard.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    padding="30px"
+    maxW="xs"
+    borderWidth="1px"
+    borderRadius="lg"
+  >
+    <Box display="flex" justifyContent="space-between">
+      <Heading size="md" marginBottom="10px">
+        {caseTitle}
+      </Heading>
+      <Button
```
Comment (bowenzhu1) [inline frontend/src/components/dashboard/CaseCard.tsx:46]: sorry for the mixup - the entire card should actually be clickable (arrow is just an indicator that the card is clickable). So on hover the card gets a shadow + the cursor shows clickable

in general, I think we can just use `onClick` of components if it's available - don't necessarily need to use `Button` for a clickable behaviour

File: frontend/src/components/dashboard/CaseCard.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+const CaseCard = ({
+  caseTitle,
+  caseLead,
+  date,
+  familyName,
+  caseTag,
+}: CaseCardProps): React.ReactElement => (
+  <Box
```
Comment (bowenzhu1) [inline frontend/src/components/dashboard/CaseCard.tsx:35]: in terms of sizing/dimensions, we want to restrict the cards to be basically squares

in that case, could we maybe try hard defining the max width and height to be based on the designs (297 and 289 are the figma values i believe? 

but spacing/arrangement and everything would be handled not in this ticket

File: frontend/src/theme/buttonStyles.tsx [hunk 3]
```diff
@@ -68,6 +68,14 @@ const Button: ComponentStyleConfig = {
         pointerEvents: "none",
       },
     },
+    withIcon: {
```
Comment (bowenzhu1) [inline frontend/src/theme/buttonStyles.tsx:71]: probably don't need this after changes

File: frontend/src/components/dashboard/CaseCard.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+export type CaseCardProps = {
+  caseTitle: string;
+  caseLead: string;
+  date: string;
+  familyName: string;
+  caseTag: string;
+};
+
+const colorChange = (value: string) => {
+  let col;
```
Comment (bowenzhu1) [inline frontend/src/components/dashboard/CaseCard.tsx:14]: nit: i think we should keep variable names as full as possible for legibility

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]
changes requested for clickable behaviour and card sizing - looks good overall!

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]
noticed some more small ui fixes to do before merging:
- we should use <Text textStyle="title-medium"> for the case card titles. the text styles should generally match what the text style is on Figma
- I think the card can be set taller vertically, and the text lines can be spaced out more. Feels off from the designs
- ^ a bit related, I think the cards should just have set width/heights. I don't think it should resize based on window size/zoom

**Review by jess94yng — APPROVED** [review]
looks good to me!

**Review by bowenzhu1 — APPROVED** [review]

**Pull request merged**